### PR TITLE
refactor: move identity and identitypopover to one folder

### LINKED ISSUE #3747: Refactor: Identity component
incremental refactor from https://github.com/kodadot/nft-gallery/issues/3714

- [x] move `Identity.vue` and `IdentityPopover` into same folder
- [x] split `Identity.vue` and `IdentityPopover` into smaller chunk
- [ ] #3760

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #3747 
- [x] move `Identity` and `IdentityPopover` into one folder
- [x] split `Identity` and `IdentityPopover` into smaller components
- [x] the rest of components only fix the eslint rules

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged the recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

File: components/shared/identity/popover/IdentityPopover.vue [hunk 1]
```diff
@@ -111,11 +61,35 @@ export default class IdentityPopover extends mixins(
...
+  }
+
+  get lastBought(): string {
+    return formatToNow(this.lastBoughtDate)
+  }
 
   public async mounted() {
+    if (this.address) {
+      this.fetchLastBought()
+    }
```
Comment (roiLeo) [inline components/shared/identity/popover/IdentityPopover.vue:89]: `async mounted` but no `await`?

File: pages/statemine/u/_id.vue [hunk 2]
```diff
@@ -28,7 +28,11 @@
             {{ $t('profile.user') }}
           </div>
           <div class="subtitle is-size-6">
-            <ProfileLink :address="id" :inline="true" showTwitter showDiscord />
+            <ProfileLink
+              :address="id"
+              :inline="true"
```
Comment (roiLeo) [inline pages/statemine/u/_id.vue:33]:
```suggestion
              inline
```
will it blend?

File: components/shared/identity/popover/IdentityPopover.vue [hunk 3]
```diff
@@ -214,6 +170,14 @@ export default class IdentityPopover extends mixins(
...
     }
   }
+
+  @Watch('address', { immediate: true })
+  protected async onAddressChanged() {
+    if (this.address) {
+      this.fetchLastBought()
+      this.fetchNFTStats()
+    }
+  }
```
Comment (roiLeo) [inline components/shared/identity/popover/IdentityPopover.vue:180]: no need await?

**Review by roiLeo — COMMENTED** [review]
Looks pretty nice, I was used to camelCase for props but you are right. ref [vuejs style guide](https://v2.vuejs.org/v2/style-guide/?redirect=true#Prop-name-casing-strongly-recommended)

I don't think we should keep `scoped` css in Identity components since they are unique class names

File: components/shared/identity/IdentityChain.vue [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  @Prop() readonly shortenedAddress!: Address
+  @Prop() readonly name!: Address
+}
+</script>
+
+<style scoped>
+.infinity-loader {
+  height: 20px;
+}
+</style>
```
Comment (roiLeo) [inline components/shared/identity/IdentityChain.vue:60]: is it needed?

File: components/shared/identity/popover/IdentityPopoverFooter.vue [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+@Component
+export default class IdentityPopoverFooter extends Vue {
+  @Prop({ required: true, default: 0 }) readonly totalCollected
+  @Prop({ required: true, default: 0 }) readonly totalCreated
+  @Prop({ required: true, default: 0 }) readonly totalSold
+}
+</script>
+
+<style scoped>
```
Comment (roiLeo) [inline components/shared/identity/popover/IdentityPopoverFooter.vue:29]:
```suggestion
<style>
```

File: components/shared/identity/popover/IdentityPopover.vue (hunk 1 above)
Comment (roiLeo) [reply components/shared/identity/popover/IdentityPopover.vue:89]: resolved?

File: components/shared/identity/IdentityChain.vue (hunk 4 above)
Comment (preschian) [reply components/shared/identity/IdentityChain.vue:60]: needed when loading, tried it with `height="20"` not working in image tag. any suggestion?

File: components/shared/identity/popover/IdentityPopover.vue (hunk 1 above)
Comment (preschian) [reply components/shared/identity/popover/IdentityPopover.vue:89]: oopps, sorry missed this one 🙏🏻

File: components/shared/identity/IdentityChain.vue (hunk 4 above)
Comment (roiLeo) [reply components/shared/identity/IdentityChain.vue:60]: It seems strange to me why this rule is necessary knowing that there is no `infinity-loader` in component, is it displaying when loaded?

Comment (preschian) [reply components/shared/identity/IdentityChain.vue:60]: oh, the line used in the line of 10 on IdentityChain.vue. I forgot to move into that component before in the previous commit 🙏🏻

**Review by roiLeo — APPROVED** [review]
✅ code lgtm

**Comment by petersopko** [comment]
nice! tried identity popover throughout the app, all seems to be working well 😄

**Review by petersopko — APPROVED** [review]
wfm

**Pull request merged**